### ISSUE TITLE
GPL-nnn As a developer I would like actions performed by users to create an audit

### ISSUE THREAD

**Issue by BenTopping** [sanger/traction-service#537]
**User story**
As a developer I would like actions performed by users to be recorded in order to make reporting easier.

**Who are the primary contacts for this story**
N/A

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] When an **successful action** is performed it creates an audit record describing what action occurred, which user peformed it, what was changed and time updated/created
- [ ] **Successful action** means an action that does not error or get rolled back. Types of actions include: importing samples, editing samples, creating/editing/deleting libraries, editing samples within a library, creating/editing runs

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

**Comment by BenTopping** [comment]
Question: Do we want audits visible in the frontend, similar to labwhere or do we just want them for reporting purposes

**Comment by TWJW-SANGER** [comment]
This sounds sensible - but given it is nearly a year and a half old I am marking as a candidate to deprecate.

**Comment by BenTopping** [comment]
Came out as a nice to have from the IsoSeq work. I imagine it will need to be implemented into traction at some point but for the time being this can probably be deprecated and then reopened / new issue when we get the resources/prioritisation to do it. @stevieing
